price-server/src/app/fees_calculator.rs

```diff
@@ -0,0 +1,69 @@
+use std::env;
+use thiserror::Error;
+
+use rust_decimal::prelude::*;
+
+#[derive(Error, Debug)]
+pub enum FeeCalculatorError {
+    #[error("Error in getting config {0}")]
+    FeeConfigError(env::VarError),
+    #[error("Error in calculating fee")]
+    CalculationError,
+    #[error("No trading amount supplied to calculate fees")]
+    NoTradingAmountAvailableError,
+}
+
+pub struct FeeCalculator {
+    base_fee_rate: Decimal,
+    immediate_fee_rate: Decimal,
+    delay_fee_rate: Decimal,
+}
+
+impl Default for FeeCalculator {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl FeeCalculator {
+    pub fn new() -> Self {
```

Comment:
Config should be injected -> don't want a dependency on the environment (or any outside state) in our domain layer.